Index: src/test/java/de/root1/simon/test/utils/BasisClass.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/BasisClass.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/BasisClass.java	(revision 0)
@@ -0,0 +1,5 @@
+package de.root1.simon.test.utils;
+
+public class BasisClass implements ExtendedBasisInterface {
+	private static final long serialVersionUID = 7059327535633065297L;
+}
Index: src/test/java/de/root1/simon/test/utils/BasisInterface.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/BasisInterface.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/BasisInterface.java	(revision 0)
@@ -0,0 +1,6 @@
+package de.root1.simon.test.utils;
+
+import de.root1.simon.SimonRemote;
+
+public interface BasisInterface extends SimonRemote {
+}
Index: src/test/java/de/root1/simon/test/utils/ExtendedBasisClass.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/ExtendedBasisClass.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/ExtendedBasisClass.java	(revision 0)
@@ -0,0 +1,5 @@
+package de.root1.simon.test.utils;
+
+public class ExtendedBasisClass extends BasisClass implements InhertedInterface {
+	private static final long serialVersionUID = -388850163638247799L;
+}
Index: src/test/java/de/root1/simon/test/utils/ExtendedBasisInterface.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/ExtendedBasisInterface.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/ExtendedBasisInterface.java	(revision 0)
@@ -0,0 +1,4 @@
+package de.root1.simon.test.utils;
+
+public interface ExtendedBasisInterface extends BasisInterface {
+}
Index: src/test/java/de/root1/simon/test/utils/InhertedClass.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/InhertedClass.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/InhertedClass.java	(revision 0)
@@ -0,0 +1,5 @@
+package de.root1.simon.test.utils;
+
+public class InhertedClass extends ExtendedBasisClass {
+	private static final long serialVersionUID = -7969016374367218992L;
+}
Index: src/test/java/de/root1/simon/test/utils/InhertedInterface.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/InhertedInterface.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/InhertedInterface.java	(revision 0)
@@ -0,0 +1,6 @@
+package de.root1.simon.test.utils;
+
+import de.root1.simon.SimonRemote;
+
+public interface InhertedInterface extends SimonRemote {
+}
Index: src/test/java/de/root1/simon/test/utils/UtilsTestCase.java
===================================================================
--- src/test/java/de/root1/simon/test/utils/UtilsTestCase.java	(revision 0)
+++ src/test/java/de/root1/simon/test/utils/UtilsTestCase.java	(revision 0)
@@ -0,0 +1,18 @@
+package de.root1.simon.test.utils;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import de.root1.simon.utils.Utils;
+
+public class UtilsTestCase {
+
+	@Test
+	public void testFindAllInterfaces() {
+		Class<?>[] interfases = Utils.findAllInterfaces(InhertedClass.class);
+		assertNotNull(interfases);
+		assertEquals(2, interfases.length);
+	}
+
+}
